os/net/link-stats.c

```diff
@@ -166,6 +166,14 @@ link_stats_packet_sent(const linkaddr_t *lladdr, int status, int numtx)
   stats->last_tx_time = clock_time();
   stats->freshness = MIN(stats->freshness + numtx, FRESHNESS_MAX);
 
+  if(status == MAC_TX_QUEUE_FULL) {
```

Comment:
Full queues were earlier reported as `MAC_TX_ERR` which did not update last timestamp and freshness (`stats->last_tx_time` and `stats->freshness`). What is the motivation of changing this behaviour?

Reply:
A good point - this PR is not meant to change that logic. Don't remember exactly why it was made this way, probably an oversight on my part.

Moving this `if` statement to line 166 should be sufficient to get the original behavior, right?

Reply:
changed the code by moving the `if` block before the freshness update logic